### ISSUE TITLE
Add core:PriorityLockTimerState to io:WindowOpenerVeluxIOComponent

### ISSUE THREAD

**Issue by BernoldNL** [Somfy-Developer/Somfy-TaHoma-Developer-Mode#109]
Is it possible to add core:PriorityLockTimerState to io:WindowOpenerVeluxIOComponent? I see it online (Overkiz), but not in developer mode.

Also, it would be great if at developer mode could have io:PriorityLockOriginatorState added. I don't see this in Overkiz either, but the actuator has a rain sensor, so it feels like it should be possible.

**Comment by qelanhari** [comment]
For my pergola, I think the originator only is présent when there is currently a lock from the rain sensor, else it does not even appear in States list.

**Comment by flhoest** [comment]
@qelanhari talking about this, how to you read the rain sensor value ? I cannot find it. I have one and it works but I have no clue how to interact with it....

**Comment by qelanhari** [comment]
There is none,if it rains, the originator is set as "rain". There is no direct state "it is raining or not". If there is no more rain, the blocker is gone.

**Comment by flhoest** [comment]
Ok, then how can I achieve this : I need to create an event based on the rain status : only do it if no rain ...
I'm a bit lost

**Comment by flhoest** [comment]
Is it the place to look at ? 

core:CommandLockLevelsState

**Comment by flhoest** [comment]
@qelanhari  ⬆️

**Comment by qelanhari** [comment]
if `core:PriorityLockTimerState` has a value above `0` (as `X`) **and** the `io:PriorityLockOriginatorState` has `rain` as value, it means that it's raining or it did a few minutes ago and the lock will clear in `X` (don't remember if it's seconds/minute or date).

**Comment by flhoest** [comment]
Thanks a lot ! I will start some testing based on that. It really helped me a lot ;)
If you are curious, I'm working on this : https://www.lets-talk-about.tech/2023/09/somfy-how-to-control-your-devices-in-php.html and another version will be released as soon as I manage the rain factor !

**Comment by qelanhari** [comment]
> @qelanhari can you send me a private message on tweeter ? @flhoest thanks ;)

-> @tankoo

**Comment by fberu-somfy** [comment]
Hello @BernoldNL ,

Yes your issue is different than the PrioryLock topic.

Regarding your points:

> When closed automatically (in this case due to rain), the core:ClosureState is not updated anymore. It should go to 100%, but it stays at the value it was. Also core:OpenClosedState stays at "open", so there is no way of knowing that it actually closed.

When raining the Velux window automatically closes but not at 100%. It closes until the "ventilation" position: the window is closed but not the top bar. core:OpenClosedState is 7% and so core:ClosureState is still "open"

> Commands setClosureAndLinearSpeed and setPositionAndLinearSpeed were added, but they don't seem to be different than setClosure (I have roller shutters that also have this and will be slower and more quiet, the Velux not so much).

You're right this is a new feature from the last 2024.3.3-10 firmware. It allows you to control the Velux Window with a low speed, in order to reduce the motor noise. To do so you can send the command setClosureAndLinearSpeed([1-100], "lowspeed") or setPositionAndLinearSpeed([1-100], "lowspeed").
Depending on the age and the version of your Velux, this command may not be available

**Comment by BernoldNL** [comment]
> Hello @BernoldNL ,
> 
> Yes your issue is different than the PrioryLock topic.
> 
> Regarding your points:
> 
> > When closed automatically (in this case due to rain), the core:ClosureState is not updated anymore. It should go to 100%, but it stays at the value it was. Also core:OpenClosedState stays at "open", so there is no way of knowing that it actually closed.
> 
> When raining the Velux window automatically closes but not at 100%. It closes until the "ventilation" position: the window is closed but not the top bar. core:OpenClosedState is 7% and so core:ClosureState is still "open"
> 
> > Commands setClosureAndLinearSpeed and setPositionAndLinearSpeed were added, but they don't seem to be different than setClosure (I have roller shutters that also have this and will be slower and more quiet, the Velux not so much).
> 
> You're right this is a new feature from the last 2024.3.3-10 firmware. It allows you to control the Velux Window with a low speed, in order to reduce the motor noise. To do so you can send the command setClosureAndLinearSpeed([1-100], "lowspeed") or setPositionAndLinearSpeed([1-100], "lowspeed"). Depending on the age and the version of your Velux, this command may not be available

Thanks for your reply. Well, in my case it's not a window, but a skylight. It does not have a ventilation position, so it does go to 100% closed. It's only about 2 years old. Mine is a CVU 100100 0225Q, originally powered from a power socket, but last year it got called back by Velux after a recall for grounding issues. It's now modified to be solar powered.

**Comment by llavorel-somfy** [comment]
Hi @niteguide what is the exact use case you want to address ?

**Comment by niteguide** [comment]
> Hi [@niteguide](https://github.com/niteguide) what is the exact use case you want to address ?

Hi @llavorel-somfy! Thanks for chiming in! 
At the moment I use the overkiz integration cloud-based. My goal would be to switch to Local. But, as also others have stated over at overkiz, there are those priority lock sensors missing in the the local API. Meaning we have not the possibility to determine if there was a storm-event. This however is necessary if Home Assistant should do something with the blinds say 2 hours after the last detected wind. Same for rain sensor. So I am waiting patiently :)

**Comment by wagaman** [comment]
@llavorel-somfy  Hi, good evening. I can see that my "issue" was marked here. I have a similar request as @niteguide that I would like to see the priority lock via the "local" mode. Thank you very much in advance.

**Comment by fberu-somfy** [comment]
Hello @wagaman ,

You issue #144 was closed as the solution is to get the "wind" error is to implement this issue and report the core:PriorityLockTimerState and core:PriorityLockOriginatorState.

We hope to get you a feedback on feasibility and planning soon.

**Comment by wagaman** [comment]
> Hello [@wagaman](https://github.com/wagaman) ,
> 
> You issue [#144](https://github.com/Somfy-Developer/Somfy-TaHoma-Developer-Mode/issues/144) was closed as the solution is to get the "wind" error is to implement this issue and report the core:PriorityLockTimerState and core:PriorityLockOriginatorState.
> 
> We hope to get you a feedback on feasibility and planning soon.

Thank you very much for getting back to me.

I will stay tuned for the update, as this feature is very important and can help prevent many dangerous activities that might occur due to unknown wind conditions.

**Comment by cook2000** [comment]
Is there a timeline for your roadmap or an information if this is something you will work on?

**Comment by niteguide** [comment]
Agree! Basically, this is some key functionality that is not supported. Somfy should really be interested in making sure people use their sensors and don’t switch to other brands just because of this aspect.

> Is there a timeline for your roadmap or an information if this is something you will work on?

**Comment by llavorel-somfy** [comment]
Hi @cook2000 @niteguide 

A patch is ready for our November release !

**Comment by llavorel-somfy** [comment]
Fixed with 2025.5.5-5 release

**Comment by BernoldNL** [comment]
I updated to 2025.5.5-7, but I don't see a difference yet. Can anyone else confirm this?

**Comment by niteguide** [comment]
@iMicknl Are there any adjustments within Overkiz necessary to get these newly exposed **local** parameters?
- Priority lock originator
- Priority lock timer

**Comment by iMicknl** [comment]
@niteguide no, we have already implemented these sensors for cloud, thus if they are available they should show up. Might require a reload of the integration / reboot of HA.

**Comment by wagaman** [comment]
You need to enable them under Sensors first. But they show unknown for me. I will observe further.

**Comment by iMicknl** [comment]
In Home Assistant you can download your diagnostics and check the JSON.

**Comment by niteguide** [comment]
I can confirm that Priority lock originator and Priority lock timer now shows up within local overkiz setup in combination with Tahoma FW 2025.5.5-7. That these sensors have state "Unknown" until something happens (e.g. wind) is normal. 
Thanks a lot for implementing!

**Comment by BernoldNL** [comment]
I still don't see it. Just tested with my Velux skylight. Opened it and then made its sensor wet. It closed automatically. But I don't see the data showing up.

@niteguide Would you mind sharing (that part of) your JSON, so I can make sure I'm not overlooking anything?

**Comment by llavorel-somfy** [comment]
Hi @BernoldNL 

Still no sign of this states ?

**Comment by BernoldNL** [comment]
> Hi [@BernoldNL](https://github.com/BernoldNL)
> 
> Still no sign of this states ?

Unfortunately not. I updated to 2025.5.5-8 yesterday. Did another test by opening the skylight and making the sensor wet. One thing that I did notice is that the window does not stay locked for long. I can reopen it again within 1 minute. The sensor can't possibly be dry again already.

These are all the states I get:
<details><summary>States</summary>
        {
          "type": 11,
          "value": [],
          "name": "core:CommandLockLevelsState"
        },
        {
          "type": 3,
          "value": "good",
          "name": "core:DiscreteRSSILevelState"
        },
        {
          "type": 1,
          "value": 86,
          "name": "core:RSSILevelState"
        },
        {
          "type": 1,
          "value": 0,
          "name": "core:ClosureState"
        },
        {
          "type": 3,
          "value": "open",
          "name": "core:OpenClosedState"
        },
        {
          "type": 11,
          "value": [],
          "name": "core:ReachedAliasesState"
        },
        {
          "type": 3,
          "value": "good",
          "name": "core:BatteryDiscreteLevelState"
        },
        {
          "type": 3,
          "value": "available",
          "name": "core:StatusState"
        },
        {
          "type": 3,
          "value": "Flatroof opener",
          "name": "core:NameState"
        }</details>

**Comment by llavorel-somfy** [comment]
> > Hi [@BernoldNL](https://github.com/BernoldNL)
> > Still no sign of this states ?
> 
> Unfortunately not. I updated to 2025.5.5-8 yesterday. Did another test by opening the skylight and making the sensor wet. One thing that I did notice is that the window does not stay locked for long. I can reopen it again within 1 minute. The sensor can't possibly be dry again already.
> 
> These are all the states I get:
> 
> States
> { "type": 11, "value": [], "name": "core:CommandLockLevelsState" }, { "type": 3, "value": "good", "name": "core:DiscreteRSSILevelState" }, { "type": 1, "value": 86, "name": "core:RSSILevelState" }, { "type": 1, "value": 0, "name": "core:ClosureState" }, { "type": 3, "value": "open", "name": "core:OpenClosedState" }, { "type": 11, "value": [], "name": "core:ReachedAliasesState" }, { "type": 3, "value": "good", "name": "core:BatteryDiscreteLevelState" }, { "type": 3, "value": "available", "name": "core:StatusState" }, { "type": 3, "value": "Flatroof opener", "name": "core:NameState" }

I am not 100% sure of what I say but it might be possible that if the status of the window changes that fast the TaHoma gateway hasn't got the time to get this state ...

**Comment by qelanhari** [comment]
I'll try on my Ondeis on my pergola directly with the API (and not via HomeAssistant) within the next few days. HomeAssistant does not get thoses states updates right now.

**Comment by BernoldNL** [comment]
> > > Hi [@BernoldNL](https://github.com/BernoldNL)
> > > Still no sign of this states ?
> > 
> > 
> > Unfortunately not. I updated to 2025.5.5-8 yesterday. Did another test by opening the skylight and making the sensor wet. One thing that I did notice is that the window does not stay locked for long. I can reopen it again within 1 minute. The sensor can't possibly be dry again already.
> > These are all the states I get:
> > States
> > { "type": 11, "value": [], "name": "core:CommandLockLevelsState" }, { "type": 3, "value": "good", "name": "core:DiscreteRSSILevelState" }, { "type": 1, "value": 86, "name": "core:RSSILevelState" }, { "type": 1, "value": 0, "name": "core:ClosureState" }, { "type": 3, "value": "open", "name": "core:OpenClosedState" }, { "type": 11, "value": [], "name": "core:ReachedAliasesState" }, { "type": 3, "value": "good", "name": "core:BatteryDiscreteLevelState" }, { "type": 3, "value": "available", "name": "core:StatusState" }, { "type": 3, "value": "Flatroof opener", "name": "core:NameState" }
> 
> I am not 100% sure of what I say but it might be possible that if the status of the window changes that fast the TaHoma gateway hasn't got the time to get this state ...

I guess that could be possible. This was the JSON during automatic closure. I can see core:ClosureState is still 0, while it should be somewhere halfway.

Any suggestion how to test this properly?

**Comment by llavorel-somfy** [comment]
You can try to launch the TaHoma app at the precise moment where your window is in priority lock state ... but again not sure at 100%

**Comment by qelanhari** [comment]
I do get the state (and so does HA), but, I think I have to try to move the slates of my pergola using the Tahoma App (or the API) and for it to refuse due to the lock, for the state to be reported.
Edit : and it goes away after the timer, until I try to move the slates again and get a new timer.

````
"states": [
            {
                "value": [],
                "type": 11,
                "name": "core:CommandLockLevelsState"
            },
            {
                "value": "normal",
                "type": 3,
                "name": "core:DiscreteRSSILevelState"
            },
            {
                "value": 66,
                "type": 1,
                "name": "core:RSSILevelState"
            },
            {
                "value": "closed",
                "type": 3,
                "name": "core:SlatsOpenClosedState"
            },
            {
                "value": {
                    "current_position": 51200
                },
                "type": 11,
                "name": "core:ManufacturerSettingsState"
            },
            {
                "value": 100,
                "type": 1,
                "name": "core:SlateOrientationState"
            },
            {
                "value": 100,
                "type": 1,
                "name": "core:SlatsOrientationState"
            },
            {
                "value": "available",
                "type": 3,
                "name": "core:StatusState"
            },
            {
                "value": "Pergola",
                "type": 3,
                "name": "core:NameState"
            },
            {
                "value": 40,
                "type": 1,
                "name": "core:Memorized1OrientationState"
            },
            {
                "value": 420,
                "type": 1,
                "name": "core:PriorityLockTimerState"
            },
            {
                "value": "rain",
                "type": 3,
                "name": "io:PriorityLockOriginatorState"
            },
            {
                "value": "comfortLevel1",
                "type": 3,
                "name": "io:PriorityLockLevelState"
            }
        ],

````

<img width="327" height="242" alt="Image" src="https://github.com/user-attachments/assets/38d4e801-0310-4e89-923f-1c5ee51c9e1f" />

**Comment by BernoldNL** [comment]
Thanks @qelanhari!

Just did some more testing and unfortunately I still don't see it. It's rainy today, so it's easier to test (don't need a ladder). When I try to open the skylight from the app I get the expected error. I have checked the setup JSON several times, but I don't see core:PriorityLockTimerState. When I check from the cloud, I do see core:PriorityLockTimerState. Strangely it has a value of 0, but apparently its presence is enough for the app(?). I don't get core:PriorityLockOriginatorState from the cloud.

I updated to firmware 2025.5.5-9 and tried again, but no luck.

**Comment by llavorel-somfy** [comment]
@BernoldNL it seems to me it is at your Velux level then no ?

**Comment by BernoldNL** [comment]
For core:PriorityLockOriginatorState I agree. But core:PriorityLockTimerState is available from the cloud, so it feels like it should also be possible to add it to Developer Mode?

**Comment by llavorel-somfy** [comment]
@BernoldNL I think it is related to your other issue #138

**Comment by BernoldNL** [comment]
But they are available on the cloud API, I just checked:

<img width="1036" height="1317" alt="Image" src="https://github.com/user-attachments/assets/dde3c10d-8a3b-44e1-aca6-7113db836239" />

**Comment by llavorel-somfy** [comment]
@BernoldNL ok but are they updated when status changes ?

**Comment by dalyand** [comment]
The cloud attributes of my awning are being updated. The priority lock originator is switching between wind and unknown. The priority lock timer is also being updated.

<img width="562" height="446" alt="Image" src="https://github.com/user-attachments/assets/57a6d8a0-a69d-46f7-a54c-c1587c87ce42" />


<img width="545" height="424" alt="Image" src="https://github.com/user-attachments/assets/6821dd3b-9e93-47c2-9a84-b26770a85aa4" />



It would be great to have these updates locally as well. To be honest, I haven't tested it in a while. I just came across it again because the issue was closed.

**Comment by wagaman** [comment]
@dalyand i am confused. I have those attributes locally as well since the recent updates for about two months now. Everything is local. One trick that made the whole thing work is to change the domain name to an actual static IP address of the TaHoma box and restart HA.

**Comment by dalyand** [comment]
@wagaman , I can confirm I have the local parameters now. Thank you verry much!
